litestar/routes/http.py

```diff
@@ -73,255 +48,20 @@ async def handle(self, scope: HTTPScope, receive: Receive, send: Send) -> None:
         Returns:
             None
         """
-        route_handler, parameter_model = self.route_handler_map[scope["method"]]
-        request: Request[Any, Any, Any] = route_handler.resolve_request_class()(scope=scope, receive=receive, send=send)
-
-        if route_handler.resolve_guards():
-            await route_handler.authorize_connection(connection=request)
-
-        response = await self._get_response_for_request(
-            scope=scope, request=request, route_handler=route_handler, parameter_model=parameter_model
-        )
-
-        await response(scope, receive, send)
-
-        if after_response_handler := route_handler.resolve_after_response():
-            await after_response_handler(request)
-
-        if form_data := scope.get("_form", {}):
-            await self._cleanup_temporary_files(form_data=cast("dict[str, Any]", form_data))
+        route_handler = self.route_handler_map[scope["method"]]
+        connection = route_handler.resolve_request_class()(scope=scope, receive=receive, send=send)
```

Comment:
Ideally this also wouldn't happen here, so the route would simply take in a `connection_class`, but that would require further config resolution in case of the HTTP handlers. I'm going to leave this up to another PR

Reply:
Why can't it happen in the handler's `handle()` method? IOW, why can't the handler's `handle()` method be an asgi app and create the connection object for themselves?

Reply:
Ideally the split would be that the routes handle the ASGI stuff and the handlers shouldn't need to be concerned with this. I know we're not there at all, but I felt like this was a good step if we want to move in that direction.
wdyt?

Reply:
Can you elaborate on what makes drawing the line for asgi interfaces there ideal? Not arguing at all - you've put a lot of thought into this so I just want to fully understand your PoV.

> the handlers shouldn't need to be concerned with this

It feels to me like a pretty thin line between handing a connection object and the individual asgi components to the handler, yet it would remove the need to book-keep the appropriate connection class type for the handler on the route side.

Reply:
You are right, this is rather arbitrary but here's my thought process so far, (keep in mind that this PR is intended to serve as a first step towards a larger refactoring goal):

In the end, I'd like routes to not be concerned with the route handler business logic at all. They simply map one or more async callables to a path (and in the case of HTTP a few methods), establish the connection and pass it to the callable.
This would also allow us to easily split off that part into a separate package if we want to do so in the future; It seems natural to me that routes are seen as a part of the routing logic, whereas handlers represent the framework side.

As for the connection vs. ASGI components, I'd also like to separate things more there, ideally only having the `ASGIResponse` classes dealing with the ASGI stuff, while everything else only handles higher-level abstractions.

I had actually already started moving the connection class config to the routes, but then I noticed it would require other changes and felt like that would be better kept as a separate PR. I could also make a PR into this branch for that, but tbh I'd like to keep these ones short lived to make them easier to keep up to date and review.

Reply:
> It seems natural to me that routes are seen as a part of the routing logic

Hard to argue with that :)

> As for the connection vs. ASGI components, I'd also like to separate things more there, ideally only having the `ASGIResponse` classes dealing with the ASGI stuff, while everything else only handles higher-level abstractions.

Let me see if I'm following correctly. Currently, we receive a "http.request" message and routing, dependency resolution, and response generation all happens within the scope of nested ASGI apps (i.e., `ExceptionHandlerMiddleware` and `route.handle`).

In this future scenario, we receive the "http.request" message and route it, then call into a non-asgi handler framework that returns a `Response` object that is naive to the server protocol (i.e., it shouldn't know or care that we are ASGI or not). We receive the protocol-naive response and use that to the protocol-aware response and call it?

> I had actually already started moving the connection class config to the routes

Still to be configurable per-handler?

Reply:
> We receive the protocol-naive response and use that to create the protocol-aware response and call it?

That about sums it up, yes!

> Still to be configurable per-handler?

Of course! Basically, we would just do e.g. `WebSocketRoute(..., connection_class=handler.resolve_websocket_class()`.

Reply:
> Of course! Basically, we would just do e.g. WebSocketRoute(..., connection_class=handler.resolve_websocket_class().

And something a little more complex for http given there could be multiple connection class types per route?

No blockers from me!

Reply:
@peterschutt I've actually changed my mind and reverted this for now. I still think it makes sense to do, but it doesn't really belong into the scope of this PR. Leaving that out makes it a bit easier to handle things down the road, and the change is easy enough to apply at some other point. I'm aiming to do this now when I'm tackling the config stuff, where it makes more sense, as it's not an intermediate step there.